symfony/plugins/orangehrmPerformancePlugin/Api/PerformanceTrackerAPI.php

```diff
@@ -0,0 +1,224 @@
+    const PARAMETER_TRACKER_NAME = 'trackerName';
+    //const PARAMETER_EMP_ID = 'empNumber';
+    const PARAM_RULE_TRACKER_NAME_MAX_LENGTH = 200;
+    const PARAMETER_REVIEWERS = 'reviewers';
+
+
+
+    /**
+     * @throws DaoException
+     * @throws NormalizeException
+     */
+    public function getAll(): EndpointResult
+    {
+        $performanceTrackerSearchParamHolder = new PerformanceTrackerSearchFilterParams();
+        $this->setSortingAndPaginationParams($performanceTrackerSearchParamHolder);
+
+        $performanceTrackerSearchParamHolder->setEmpNumber(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_EMPLOYEE_NUMBER
+            )
+        );
+        $performanceTrackers = $this->getPerformanceTrackerService()->getPerformanceTrackerDao()->getPerformanceTrackList($performanceTrackerSearchParamHolder);
+        $count =$this->getPerformanceTrackerService()->getPerformanceTrackerDao()->getPerformanceTrackerCount($performanceTrackerSearchParamHolder);
+        return new EndpointCollectionResult(
+            PerformanceTrackerModel::class,
+            $performanceTrackers,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL=> $count])
+        );
+    }
+
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    CommonParams::PARAMETER_EMP_NUMBER,
+                    new Rule(Rules::POSITIVE)
+                ),
+            ),
+            ...$this->getSortingAndPaginationParamsRules(PerformanceTrackerSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    public function create(): EndpointResult
+    {
+       $performanceTracker = new PerformanceTracker();
+       $reviewers =$this->getRequestParams()->getArray(
+           RequestParams::PARAM_TYPE_BODY,
+           self::PARAMETER_REVIEWERS
+       );
+       $this->setPerformanceTrackerParams($performanceTracker,1);
+       $this->getPerformanceTrackerService()->getPerformanceTrackerDao()->savePerformanceTracker($performanceTracker,$reviewers);
+       return new EndpointResourceResult(PerformanceTrackerModel::class, $performanceTracker);
+    }
+
+    private function setPerformanceTrackerParams(PerformanceTracker $performanceTracker ,int $num): void
+    {
+        $performanceTracker->setTrackerName(
+            $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_TRACKER_NAME)
+        );
+        $empNumber =$this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_BODY, CommonParams::PARAMETER_EMP_NUMBER);
+        $performanceTracker->getDecorator()->setEmployeeByEmpNumber($empNumber);
+        $addedEmp =$this->getAuthUser()->getEmpNumber();
+        $performanceTracker->getDecorator()->setAddedByByEmpNumber($addedEmp);
```

Comment:
you can directly pass like this also
```suggestion
        $performanceTracker->getDecorator()->setAddedByByEmpNumber($this->getAuthUser()->getEmpNumber());
```

Reply:
changed to this